app/models/restaurant.rb

```diff
@@ -14,4 +14,12 @@ class Restaurant
   validates :url, presence: true, uniqueness: true, format: VALID_URL_REGEX
 
   scope :random, -> { skip(rand(self.count)) }
+
+  def self.search(keyword)
+    if keyword
+      where( name: /#{keyword}/ )
```

Comment:
@yukimurasawa ユーザーが自由に入力するパラメータをそのままクエリに入れると、SQLインジェクションの危険性があります。
こういった場合は、文字列をエスケープしてから渡すようにしましょう！
```suggestion
      where( name: /#{Regexp.escape(keyword)}/ )
```

（もし↑このコードでうまくいかなかったら `::Regexp.escape(keyword)` とした方がいいかも）

セキュリティ観点で安心安全なシステムにすることは重要なので、覚えておこう！
SQL インジェクションや `Regexp.escape` については自分で詳しく調べてみてください〜！